src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapFandom.java

```diff
@@ -1,206 +1,220 @@
-
-        videoFandomPage(slotName).verifyVideoClosesAfterTapOnCloseButton(videoFanTakeover);
-    }
-
-    @NetworkTrafficDump(useMITM = true)
-    @InBrowser(
-            browser = Browser.CHROME,
-            emulator = Emulator.GOOGLE_NEXUS_5
-    )
-    @Test(
-            dataProviderClass = FandomAdsDataProvider.class,
-            dataProvider = "vuapPageMobile",
-            groups = {"AdsVuapTimeProgressMobileFandom"}
-    )
-    public void adsVuapTimeProgressingFandomMobile(String pageType, String pageName, String slotName,
-                                                   String iframeId, String videoUrl) throws InterruptedException {
-        networkTrafficInterceptor.startIntercepting();
-        AdsFandomObject fandomPage = loadPage(pageName, pageType);
-        VideoFanTakeover videoFanTakeover = prepareSlot(slotName, iframeId, fandomPage);
-        fandomPage.wait.forSuccessfulResponseByUrlPattern(networkTrafficInterceptor, videoUrl);
-
-        videoFandomPage(slotName).verifyIsVideoTimeProgresingOnMobile(networkTrafficInterceptor, videoFanTakeover);
-    }
-
-    @NetworkTrafficDump(useMITM = true)
-    @InBrowser(
-            browser = Browser.CHROME,
-            emulator = Emulator.GOOGLE_NEXUS_5
-    )
-    @Test(
-            dataProviderClass = FandomAdsDataProvider.class,
-            dataProvider = "vuapPageMobile",
-            groups = {"AdsVuapFandomMobile", "AdsVuapVideoPauseFandomMobile"}
-    )
-    public void adsVuapVideoPauseFandomMobile(String pageType, String pageName, String slotName,
-                                              String iframeId, String videoUrl) throws InterruptedException {
-        networkTrafficInterceptor.startIntercepting();
-        AdsFandomObject fandomPage = loadPage(pageName, pageType);
-        VideoFanTakeover videoFanTakeover = prepareSlot(slotName, iframeId, fandomPage);
-        fandomPage.wait.forSuccessfulResponseByUrlPattern(networkTrafficInterceptor, videoUrl);
-
-        videoFandomPage(slotName).verifyIsVideoPausedOnMobile(networkTrafficInterceptor, videoFanTakeover);
-    }
-
-    private VideoFanTakeover prepareSlot(String slotName, String iframeId, AdsFandomObject fandomPage) {
-        fandomPage.triggerOnScrollSlots();
-        VideoFanTakeover videoFanTakeover = new VideoFanTakeover(driver, iframeId, AdsFandomContent.getGptSlotSelector(slotName));
-        fandomPage.scrollToSlot(AdsFandomContent.getGptSlotSelector(slotName));
-        return videoFanTakeover;
-    }
-
-    private FandomVideoFanTakeover videoFandomPage(String slotName) {
-        return new FandomVideoFanTakeover(driver, slotName);
-    }
+public class TestAdsVuapFandom extends AdsFandomTestTemplate {
+  private static final long MAX_MOVIE_DURATION = 40L;
+  private static final int DESKTOP_VIDEO_TRIGGER_AREA = 2;
+  private static final int MOBILE_VIDEO_TRIGGER_AREA = 3;
+  private static final int REDIRECT_AREA_TRIGGER = 1; // DESKTOP & MOBILE
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3ATrailingCommentCheck)